src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
Yup. I seem to have made a fatal incorrect assumption about concurrency (again :P)